.travis.yml

```diff
@@ -3,42 +3,67 @@ language: python
 matrix:
   include:
   - python: 3.5
-    env: export FAIL_ON_EXTERNAL_DEPRECATION='False'
+    env: export FAIL_ON_EXTERNAL_DEPRECATION='False'; MINIMAL_ENV='False'
   - python: 3.5
-    env: export FAIL_ON_EXTERNAL_DEPRECATION='True'
+    env: export FAIL_ON_EXTERNAL_DEPRECATION='True'; MINIMAL_ENV='False'
+  - python: 3.5
+    env: export FAIL_ON_EXTERNAL_DEPRECATION='False'; MINIMAL_ENV='True'
   - os: osx
     language: generic
     python: 3.5
-    env: export FAIL_ON_EXTERNAL_DEPRECATION='False'
+    env: export FAIL_ON_EXTERNAL_DEPRECATION='False'; MINIMAL_ENV='False'
   allow_failures:
-  - env: export FAIL_ON_EXTERNAL_DEPRECATION='True'
+  - env: export FAIL_ON_EXTERNAL_DEPRECATION='True'; MINIMAL_ENV='False'
 
 before_install:
-- if [[ $TRAVIS_OS_NAME == 'osx' ]] ; then curl "https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh" -o miniconda.sh; else wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh; fi
-- chmod +x miniconda.sh
-- "./miniconda.sh -b -p $HOME/miniconda"
-- export PATH="$HOME/miniconda/bin:$PATH"
-- hash -r
-- conda update --yes conda
-
+  - if [[ $TRAVIS_OS_NAME == 'osx' ]] ; then
+      curl "https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh" -o miniconda.sh;
+    else wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
+    fi
+  - if [[ $MINIMAL_ENV == 'False' ]] ; then
+      chmod +x miniconda.sh;
+      ./miniconda.sh -b -p $HOME/miniconda;
+      export "PATH=$HOME/miniconda/bin:$PATH";
+      hash -r;
+      conda update --yes conda;
+    fi
 install:
-  - DEPS="nose numpy scipy matplotlib ipython h5py sympy scikit-learn dill natsort setuptools scikit-image cython lxml ipyparallel"
-  - conda create -n testenv --yes $DEPS
-  - source activate testenv
-  - conda install pip
-  - pip install coverage coveralls traits traitsui requests tqdm
-  - python setup.py install
+  - if [[ $MINIMAL_ENV == 'False' ]] ; then
+      DEPS="nose numpy scipy matplotlib ipython h5py sympy scikit-learn dill natsort setuptools scikit-image cython lxml ipyparallel";
+      conda create -n testenv --yes $DEPS;
+      source activate testenv;
+      conda install pip;
+      pip install coverage coveralls traits traitsui requests tqdm;
+      python -c "import sys; print(sys.executable)";
+      python setup.py install;
+    else pip3 install .;
```

Comment:
Imho think `pip install coverage coveralls...` should be  the same moved to `script` there whe it is needed. This way time is optimized, so if installation fails, no time is spent then for installing stuff it is not going to be used. I understand the urge to have it organized into travis present `before_install`, `install`, `script`... sections, but build/testing is all ready taking too long time, You can get nap while waiting for travis to finish...